File: Back-laravel/app/Http/Requests/Auth/RegisterRequest.php
<?php

namespace App\Http\Requests\Auth;

use App\Http\Requests\ApiRequest;

class RegisterRequest extends ApiRequest
{
    public function rules(): array
    {
        return [
            'email'     => ['required', 'email', 'max:255', 'unique:users,email'],
            'password'  => ['required', 'string', 'min:8', 'max:100', 'regex:/[A-Z]/', 'regex:/[a-z]/', 'regex:/[0-9]/', 'regex:/[^A-Za-z0-9]/'],
            'firstName' => ['required', 'string', 'min:2', 'max:100'],
            'lastName'  => ['required', 'string', 'min:2', 'max:100'],
            'phone'     => ['nullable', 'string', 'regex:/^\+?[1-9]\d{1,14}$/', 'unique:users,phone'],
            'role'      => ['nullable', 'string', 'in:CANDIDATE,RECRUITER'],
        ];
    }
}

File: Back-ASPNET/src/RecruitmentAuth.Application/DTOs/Auth/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RecruitmentAuth.Application.DTOs.Auth;

public class RegisterRequest
{
    [Required] public string FirstName { get; set; } = string.Empty;
    [Required] public string LastName { get; set; } = string.Empty;
    [Required, EmailAddress] public string Email { get; set; } = string.Empty;
    [Required, MinLength(8)] public string Password { get; set; } = string.Empty;
    public string? Phone { get; set; }
    /// <summary>CANDIDATE | RECRUITER</summary>
    public string Role { get; set; } = "CANDIDATE";
}
